cylc/uiserver/authorise.py

```diff
@@ -387,6 +385,20 @@ def return_site_auth_defaults_for_access_user(
         defaults.discard("")
         return defaults
 
+    def _get_groups(self, user: str) -> List:
+        """Allows get groups to use self.logger if something goes wrong.
+
+        Added to provide a single interface for get_groups to this class, to
+        avoid having to pass the logger to get_groups (and methods it calls).
+        """
+        good_groups, bad_groups = get_groups(user)
+        if bad_groups:
+            self.log.warning(
+                f'{user} has the following invalid groups in their profile '
+                f'{bad_groups}'
```

Comment:
Worth nothing here that these groups will be ignored for authorisation purposes.

Reply:
I was struggling to write these in a way I was happy with.